client/state/sharing/keyring/selectors.js

```diff
@@ -49,6 +56,50 @@ export function getUserConnections( state, userId ) {
 }
 
 /**
+ * Given a service, returns a flattened array of all possible accounts for the
+ * service for which a connection can be created.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {String} service The name of the service to check
+ * @return {Array}          Flattened array of all possible accounts for the service
+ */
+export function getAvailableExternalConnections( state, service ) {
```

Comment:
How would you describe the return value of this function? Is it an array of Keyring connections? Publicize connections? Since they're more of a custom structure, this may be why I had previously named the function as "Accounts", not "Connections".

Also, as an aside, the placement of selectors between `sharing/keying` and `sharing/publicize` seems difficult - should this be in Publicize, for example? This was part of the reason for creating a nested `sharing` directory, that we could have selectors live in `sharing/selectors.js` to reduce confusion when accessing both parts of the state.